packages/syft/src/syft/core/node/common/node_service/user_manager/new_user_messages.py

```diff
@@ -0,0 +1,385 @@
+from nacl.signing import VerifyKey
+from pydantic import EmailStr
+from typing_extensions import final
+
+# relative
+from .....common.serde.serializable import serializable
+from ....abstract.node_service_interface import NodeServiceInterface
+from ....domain.domain_interface import DomainInterface
+from ....domain.registry import DomainMessageRegistry
+from ...exceptions import AuthorizationError
+from ...exceptions import MissingRequestKeyError
+from ...exceptions import UserNotFoundError
+from ...node_table.utils import model_to_json
+from ...permissions.permissions import BasePermission
+from ...permissions.permissions import BinaryOperation
+from ...permissions.permissions import UnaryOperation
+from ...permissions.user_permissions import IsNodeDaaEnabled
+from ...permissions.user_permissions import UserCanCreateUsers
+from ...permissions.user_permissions import UserCanTriageRequest
+from ...permissions.user_permissions import UserIsOwner
+from ..generic_payload.syft_message import NewSyftMessage as SyftMessage
+from ..generic_payload.syft_message import ReplyPayload
+from ..generic_payload.syft_message import RequestPayload
+
+
+@serializable(recursive_serde=True)
+@final
+class CreateUserMessage(SyftMessage, DomainMessageRegistry):
+
+    # Pydantic Inner class to define expected request payload fields.
+    class Request(RequestPayload):
+        """Payload fields and types used during a User Creation Request."""
+
+        email: EmailStr
+        password: str
+        name: str
+        role: Optional[str] = "Data Scientist"
+        institution: Optional[str]
+        website: Optional[str]
+        budget: Optional[float] = 0.0
+        daa_pdf: Optional[bytes] = b""
+
+    # Pydantic Inner class to define expected reply payload fields.
+    class Reply(ReplyPayload):
+        """Payload fields and types used during a User Creation Response."""
+
+        message: str = "User created successfully!"
+
+    request_payload_type = (
+        Request  # Converts generic syft dict into a CreateUserMessage.Request object.
+    )
+    reply_payload_type = (
+        Reply  # Creates a proper Reply payload message structure as a response.
+    )
+
+    def run(  # type: ignore
+        self, node: DomainInterface, verify_key: Optional[VerifyKey] = None
+    ) -> ReplyPayload:  # type: ignore
+        """Validates the request parameters and creates a new user.
+
+        Args:
+            node (DomainInterface): Domain interface node.
+            verify_key (Optional[VerifyKey], optional): User signed verification key. Defaults to None.
+
+        Raises:
+            MissingRequestKeyError: If the required request fields are missing.
+            AuthorizationError: If user already exists for given email address.
+
+        Returns:
+            ReplyPayload: Message on successful user creation.
+        """
+
+        # Check if email/password fields are empty
+        if not getattr(self.payload, "email", "") or not getattr(
```

Comment:
Do we need to use `getattr` here, or can we use the . syntax on payload since it's a proper PyDantic type? What does PyDantic provide for Optional, will it be None and evaluate to False?

Reply:
Actually, we don't even need this check anymore since `email` and `password` fields are mandatory by pydantic payload class. So I just removed this check and added a `try/catch` code in `SyftMessage.payload` to check if the payload has been built properly (capturing `PydanticValidationError` and raising a `BadPayloadException`  otherwise).

Reply:
Fixed by: [cb9b859](https://github.com/OpenMined/PySyft/pull/6274/commits/cb9b859fbfe0a3461e9a8a1e3a20f244e5200316), [6a48881](https://github.com/OpenMined/PySyft/pull/6274/commits/6a4888162114da735aa75930cba4973703c5aa75),[0b9991d](https://github.com/OpenMined/PySyft/pull/6274/commits/0b9991d216524a64db56d30bf4fabf26ddbe0663)